src/App.tsx

```diff
@@ -8,7 +8,65 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { TodoStatus } from './types/TodoStatus';
+
+import { getTodos } from './api';
+
+const getFilteredTodos = (todos: Todo[], query: string, filterBy: FilterBy) => {
+  let newTodos = [...todos];
+
+  if (query) {
+    const newQuery = query.toLowerCase().trim();
+
+    newTodos = newTodos.filter(todo => todo.title.toLowerCase().includes(newQuery));
+  }
+
+  switch (filterBy) {
+    case TodoStatus.Active:
+      return newTodos.filter(todo => !todo.completed);
+
+    case TodoStatus.Completed:
+      return newTodos.filter(todo => todo.completed);
+
+    default:
+      break;
+  }
+
+  return newTodos;
+};
+
```

Comment:
1. Let's move this function to the `helpers` file
2. Do you need to copy the array, when you use only `filter()` method?
3. Use 1 filter instead and move the `switch` case logic and `includes` to the callback

```suggestion
const getFilteredTodos = (todos: Todo[], query: string, filterBy: FilterBy) => {
  const newQuery = query.toLowerCase().trim();
  
  return todos.filter(todo => {
     const isTitleIncludeQuery = todo.title.toLowerCase().includes(newQuery);
     switch (filterBy) {
      case TodoStatus.All:
        return isTitleIncludeQuery
     
      case TodoStatus.Active:
        return isTitleIncludeQuery && ...
      ...
  
      default:
        break;
    }
  })
};

```